DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs

```diff
@@ -248,44 +252,26 @@ private IQueryable<DeviceModel> SearchDeviceList(IQueryable<DeviceModel> deviceL
             // look for all devices that contain the search value in one of the DeviceProperties Properties
             return deviceList.Where(filter).AsQueryable();
         }
-       
+
         private bool SearchTypePropertiesForValue(DeviceModel device, string search)
         {
-            DeviceProperties devProps = null;
-
             // if the device or its system properties are null then
             // there's nothing that can be searched on
-            if ((device == null) ||
-                ((devProps = DeviceSchemaHelper.GetDeviceProperties(device)) == null))
-            {
-                return false;
-            }
-
-            try
-            {
-                devProps = DeviceSchemaHelper.GetDeviceProperties(device);
-            }
-            catch (DeviceRequiredPropertyNotFoundException)
-            {
-                devProps = null;
-            }
-
-            if (devProps == null)
+            if (device?.DeviceProperties == null)
```

Comment:
this is the first time I'm seeing this :) device?.DeviceProperties 
does it do if(device == null || device.DeviceProperties ==null)